pages/en/indexing/operating-graph-node.mdx

```diff
@@ -0,0 +1,397 @@
+
+> TODO: details on specific recoverable issues (database unavailable? poisoned by previous?)
+
+### Block and call cache
+
+Graph Node caches certain data in the store in order to save refetching from the provider. Blocks are cached, as are the results of `eth_calls` (the latter being cached as of a specific block). This caching can dramatically increase indexing speed during "resyncing" of a slightly altered subgraph.
+
+However in some instances, if an Ethereum node has provided incorrect data for some period, that can make its way into the cache, leading to incorrect data or failed subgraphs. In this case indexers can use `graphman` to clear the poisoned cache, and then rewind the affected subgraphs, which will then fetch fresh data from the (hopefully) healthy provider.
+
+If a block cache inconsistency is suspected, such as a tx receipt missing event:
+
+1. `graphman chain list` to find the chain name.
+2. `graphman chain check-blocks <CHAIN> by-number <NUMBER>` will check if the cached block matches the provider, and print the diff delete the block from the cache if it doesn’t.
+    1. If there is a difference, it may be safer to truncate the whole cache with `graphman chain truncate <CHAIN>`.
+    2. If the block is matches the provider, then the issue can be debugged directly against the provider.
+
+### Querying issues and errors
+
+Once a subgraph has been indexed, indexers can expect to serve queries via the subgraph's dedicated query endpoint. If the indexer is hoping to serve significant query volume, a dedicated query node is recommended, and in case of very high query volumes, indexers may want to configure replica shards so that queries don't impact the indexing process.
+
+However even with a dedicated query node and replicas, certain queries can take a long time to execute, and in some cases increase memory usage and negatively impact the query time for other users.
+
+There is not one "silver bullet", but a range of tools for preventing, diagnosing and dealing with slow queries.
+
+#### Query caching
+
+Graph Node caches GraphQL queries by default, which can significantly reduce database load. This can be further configured with the `GRAPH_QUERY_CACHE_BLOCKS` and `GRAPH_QUERY_CACHE_MAX_MEM` settings - read more [here](https://github.com/graphprotocol/graph-node/blob/master/docs/environment-variables.md#graphql-caching).
+
+#### Analysing queries
+
+Problematic queries most often surface in one of two ways. In some cases, users themselves report that a given query is slow. In that case the challenge is to diagnose the reason for the slowness - whether it is a general issue, or specific to that subgraph or query. And then of course to resolve it, if possible.
+
+In other cases, the trigger might be high memory usage on a query node, in which case the challenge is first to identify the query causing the issue.
+
+> TODO: how to monitor memory usage / query node saturation?
```

Comment:
@kaiwetlesen would be interested in understanding your setup for monitoring memory usage and if that could be generally useful